lib/services/local/access_list.go

```diff
@@ -214,14 +214,19 @@ func (a *AccessListService) runOpWithLock(ctx context.Context, accessList *acces
 	validateAccessList := func() error {
 		var err error
 
+		existingAccessList, getResourceErr := a.service.GetResource(ctx, accessList.GetName())
+		if getResourceErr != nil && !trace.IsNotFound(getResourceErr) {
+			return trace.Wrap(err)
+		}
```

Comment:
Do you really need to move this out of the "update" case below? I think you can just assign `var existingAccessList *accesslist.AccessList` and use it below - it will be `nil` in case of upsert and there will be no extra backend call.

In fact, there is _already_ `var existingList *accesslist.AccessList` variable declared above (line 207), why do we need another one? `existingList` vs `existingAccessList` is quite confusing.